src/main/javascript/service/cat-api-service.js

```diff
@@ -323,6 +323,20 @@ function CatApiServiceProvider() {
             function $getCatApiService($http, catConversionService) {
             var catApiService = {};
 
+            /**
+             * This method allows to define (dynamic) endpoints after the configuration phase.
+             * @param {object} [settings] a new {EndpointConfig} will be created with the given settings
+             * @returns {CatApiEndpoint}
+             */
+            catApiService.$endpoint = function (settings) {
+                var name = 'dynamic-endpoint';
+                if (!_.isUndefined(settings)) {
+                    return new CatApiEndpoint(_urlPrefix, new EndpointConfig(name, settings), $http, catConversionService);
```

Comment:
Also good. I'll make the name optional though.. It's dynamic. That means the developer (if forced) would have to assemble an extra string for the name just for making them unique.